sdks/python/apache_beam/options/pipeline_options_validator.py

```diff
@@ -23,6 +23,7 @@
 
 import logging
 import re
+import validators
```

Comment:
are we comfortable to introduce this new package? I recommend we should use pydantic: https://docs.pydantic.dev/latest/usage/types/urls/

Reply:
Thinking it over, I decided to do a very basic validation using urllib. It isn't worth bringing in new packages for just this functionality.

Reply:
make sense. Thanks.

Reply:
another question: do we know how the Java SDK handle this option now?

Reply:
Java doesn't validate the url. It also optionally uses an ENV variable.